packages/core/src/envelope.ts

```diff
@@ -128,20 +132,22 @@ function createEventEnvelopeHeaders(
     ...(event.type === 'transaction' &&
       event.contexts &&
       event.contexts.trace && {
```

Comment:
Do we only want to add trace context data here if there's already trace context data on the event coming into this function? If so, is that because if we don't already have trace context, we don't have a trace id, and without a trace id, the rest is pointless? (If so, mind adding a comment to that effect, please?)

Reply:
Yes, I think that's exactly it. I added this condition in https://github.com/getsentry/sentry-javascript/pull/5128/commits/80dad8c5341218a1b6ebafbaa249b5964c11dc5f after reading up on what Relay [expects to find](https://github.com/getsentry/relay/blob/b23b8d3b2360a54aaa4d19ecae0231201f31df5e/relay-sampling/src/lib.rs#L685-L707) in the header. It expects a `trace_id` which we get from `event.contests.trace.trace_id`.

Reply:
Right, makes sense.